doc/tutorials/chapter_6/00_combinational_logic.md

```diff
@@ -0,0 +1,383 @@
+
+So, how about `switch...case`? Well, its pretty much the same. Let see the syntax in `Case`.
+
+## Case
+
+ROHD supports Case statements, including priority and unique flavors, which are implemented in the same way as SystemVerilog. For example:
+
+### Start by declaring a Case
+
+`Case([])`: In a Case statement, each item in the `items` list is checked against the `expression`. If a matching item is found, its associated code block is executed. If none of the items match, the code block associated with the [defaultItem] is executed instead. This allows you to handle different scenarios in your program based on the value of the [expression].
+
+```dart
+Case(
+  Logic expression,
+  List<CaseItem> items, {
+  List<Conditional>? defaultItem,
+})
+```
+
+### Add Expressions
+
+`Logic expression`: `Case` expressions are made up of logical values. If you have multiple signals that you want to combine into one expression, you can use the `swizzle()` function. This function allows you to create a new expression by selecting individual elements from existing expressions and combining them in a specified order. By using `swizzle()`, you can create complex expressions that represent the behavior of your circuit.
+
+```dart
+Case(
+  // swizzle signal a and b as expression
+  [b, a].swizzle(),
+)
+```
+
+### Add Case Items
+
+`CaseItem(Logic value, List<Conditional> then)`: The `CaseItem` function takes a `Logic` value as a condition. If the condition is met, the `then` condition will be executed. The `then` condition is a list of actions to be carried out during the simulation.
+
+```dart
+List<CaseItem>
+```
+
+Remember that, the `CaseItem` should be in the List of the `List<CaseItem>` reside in `Case`.
+
+```dart
+Case(
+  // swizzle signal a and b as expression
+  [b, a].swizzle(),
+  [
+    // Case Items here
+    CaseItem(Const(LogicValue.ofString('01')), [
+      c < 1,
+      d < 0
+    ]),
+  ], {
+  List<Conditional>? defaultItem,
+})
+```
+
+### Add default Items
+
+`List<Conditional>? defaultItem` : `defaultItem` conditions will be executed if none of the expression matched.
+
+```dart
+Case(
+  // swizzle signal a and b as expression
+  [b, a].swizzle(),
+  [
+    // Case Items here
+    CaseItem(Const(LogicValue.ofString('01')), [
+      c < 1,
+      d < 0
+    ]),
+  ], {
+  List<Conditional>? defaultItem,
+})
+```
+
+### Encapsulate Case into a Combinational
+
+```dart
+Combinational([
+  Case([b,a].swizzle(), [
+      CaseItem(Const(LogicValue.ofString('01')), [
+        c < 1,
+        d < 0
+      ]),
+      CaseItem(Const(LogicValue.ofString('10')), [
+        c < 1,
+        d < 0,
+      ]),
+    ], defaultItem: [
+      c < 0,
+      d < 1,
+    ],
+    conditionalType: ConditionalType.Unique
+  )
+]);
+```
+
+Yes, that it for `Case` in ROHD.
```

Comment:
Maybe a mention of `CaseZ` would be helpful, even if not a full walk-through and example